channeldb/graph.go

```diff
@@ -1061,24 +1061,34 @@ func (c *ChannelGraph) DisconnectBlockAtHeight(height uint32) ([]*ChannelEdgeInf
 
 		// Scan from chanIDStart to chanIDEnd, deleting every
 		// found edge.
+		// NOTE: we must delete the edges after the cursor loop, since
+		// modifying the bucket while traversing is not safe.
+		var keys [][]byte
 		cursor := edgeIndex.Cursor()
 		for k, v := cursor.Seek(chanIDStart[:]); k != nil &&
 			bytes.Compare(k, chanIDEnd[:]) <= 0; k, v = cursor.Next() {
 
+			key := make([]byte, len(k))
+			copy(key[:], k[:])
+			keys = append(keys, key)
+
 			edgeInfoReader := bytes.NewReader(v)
 			edgeInfo, err := deserializeChanEdgeInfo(edgeInfoReader)
 			if err != nil {
 				return err
 			}
+
+			removedChans = append(removedChans, &edgeInfo)
```

Comment:
Any reason for moving this up here?

Reply:
It's not really moved, it just looks that way because the `delChannelEdge` calls between are moved down.